Faq cd (#6306)
harness/developer-hub: kb/continuous-delivery/continuous-delivery-faqs.md

```

This error occurs if using a Monitored Service with an incorrectly configured Change Source. To fix this, please ensure that the Change Source has a ServiceId defined properly.

#### How do I get the index of a looping strategy?

> You can use the `<+strategy.identifierPostFix>` expression to get the index of a strategy. More information can be found in the [identifierPostFix expressions](https://developer.harness.io/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism/#identifierpostfix-expressions) documentation

Reviewer: ```suggestion
You can use the `<+strategy.identifierPostFix>` expression to get the index of a strategy. More information, go to [identifierPostFix expressions](https://developer.harness.io/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism/#identifierpostfix-expressions).
```

#### How do I use Maps in Harness Expressions?

> You can convert the Map to JSON and use the `<+json.select()>` function in order to achieve this. More information can be found in the [JSON and XML Functors](https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/json-and-xml-functors/) documentation

Reviewer: ```suggestion
You can convert the Map to JSON and use the `<+json.select()>` function in order to achieve this. For more information, go to [JSON and XML Functors](https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/json-and-xml-functors/).
```

#### How do I get the action of my Git trigger in an expression?

> Git triggers use webhooks, and webhooks usually have a payload that you can utilize. The action used to trigger the webhook should also be included in the Git oayload so that you can reference the action using the `<+trigger.payload.action>` expression.

Reviewer 2: ```suggestion
Git triggers use webhooks, and webhooks usually have a payload that you can utilize. The action used to trigger the webhook must be included in the Git payload so that you can reference the action using the `<+trigger.payload.action>` expression.
```

> #### How do I get the index of a looping strategy?
> You can use the `<+strategy.identifierPostFix>` expression to get the index of a strategy. More information can be found in the [identifierPostFix expressions](https://developer.harness.io/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism/#identifierpostfix-expressions) documentation
> 
> #### How do I use Maps in Harness Expressions?
> You can convert the Map to JSON and use the `<+json.select()>` function in order to achieve this. More information can be found in the [JSON and XML Functors](https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/json-and-xml-functors/) documentation
> 
> #### How do I get the action of my Git Trigger in an expression?
> Git Triggers use Webhooks and where there's a webhook there's usually a payload you can utilize. The action used to trigger the Webhook should also be included in the Git Payload so you can reference the action using `<+trigger.payload.action>`.

Reviewer 2: ```suggestion
#### How do I get the index of a looping strategy?
You can use the `<+strategy.identifierPostFix>` expression to get the index of a strategy. For more information, go to the [identifierPostFix expressions](https://developer.harness.io/docs/platform/pipelines/looping-strategies/looping-strategies-matrix-repeat-and-parallelism/#identifierpostfix-expressions) documentation.

#### How do I use maps in Harness expressions?
You can convert map to JSON and use the `<+json.select()>` function to achieve this. For more information, go to [JSON and XML Functors](https://developer.harness.io/docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/json-and-xml-functors/) documentation.

#### How do I get the action of my Git trigger in an expression?
Git triggers use webhooks, and webhooks usually have a payload that you can utilize. The action used to trigger the webhook should also be included in the Git oayload so that you can reference the action using the `<+trigger.payload.action>` expression.
```
